src/main/java/main/internet/shop/dao/jdbcimpl/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,170 @@
+package main.internet.shop.dao.jdbcimpl;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import main.internet.shop.dao.OrderDao;
+import main.internet.shop.exception.DataProcessingException;
+import main.internet.shop.lib.Dao;
+import main.internet.shop.model.Order;
+import main.internet.shop.model.Product;
+import main.internet.shop.utils.ConnectionUtils;
+
+@Dao
+public class OrderDaoJdbcImpl implements OrderDao {
+    @Override
+    public List<Order> getUserOrders(Long userId) {
+        List<Order> orders = new ArrayList<>();
+        String query = "SELECT * FROM orders WHERE user_id = ? AND deleted = false";
+        try (Connection connection = ConnectionUtils.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, userId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                orders.add(extractOrderFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get orders from DB with user Id = "
+                    + userId, e);
+        }
+        for (Order order : orders) {
+            order.setProducts(extractProductsFromOrder(order.getId()));
+        }
+        return orders;
+    }
+
+    @Override
+    public Order create(Order order) {
+        String query = "INSERT INTO orders (user_id) VALUES (?)";
+        try (Connection connection = ConnectionUtils.getConnection();
+                PreparedStatement statement = connection.prepareStatement(
+                        query, PreparedStatement.RETURN_GENERATED_KEYS)) {
+            statement.setLong(1, order.getUserId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                order.setId(resultSet.getLong(1));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create order with id = " + order.getId(), e);
+        }
+        return insertProducts(order);
+    }
+
+    @Override
+    public Optional<Order> get(Long id) {
+        String query = "SELECT * FROM orders WHERE id = ?";
+        try (Connection connection = ConnectionUtils.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (!resultSet.next()) {
+                return Optional.empty();
+            }
+            Order order = extractOrderFromResultSet(resultSet);
+            order.setProducts(extractProductsFromOrder(order.getId()));
```

Comment:
close statement before using another one

Reply:
let's either move this after catch block or the send the same connection  object to method in order not to open connection inside other connection